File: docs/guides/php/src/mcmSearchWithout.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;
use Algolia\AlgoliaSearch\Model\Search\SearchParamsObject;

$getAppIDFor = function ($user) {
    return ''; // Implement your own logic here
};

$getIndexingApiKeyFor = function ($user) {
    return ''; // Implement your own logic here
};

// Fetch from your own data storage and with your own code
// the associated application ID and API key for this user
$appID = $getAppIDFor('user42');
$apiKey = $getIndexingApiKeyFor('user42');

$client = SearchClient::create($appID, $apiKey);

$searchParams = (new SearchParamsObject())
    ->setQuery('<YOUR_SEARCH_QUERY>')
    ->setFacetFilters(['user:user42', 'user:public'])
;

$client->searchSingleIndex(
    '<YOUR_INDEX_NAME>',
    $searchParams,
);

File: docs/guides/ruby/mcmSearchWithout.rb
import(time)

require "algolia"

def get_app_id_for(_user)
  # Implement your own logic here
  ""
end

def get_indexing_api_key_for(_user)
  # Implement your own logic here
  ""
end

app_id = get_app_id_for("user42")
api_key = get_indexing_api_key_for("user42")

client = Algolia::SearchClient.create(app_id, api_key)

search_params = Algolia::Search::SearchParamsObject.new(
  query: "<YOUR_SEARCH_QUERY>",
  facet_filters: %w[user:user42 user:public]
)

client.search_single_index("<YOUR_INDEX_NAME>", search_params)
